ed25519/options.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ed25519
+
+import (
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+)
+
+const (
+	ConfigPrivKey     = "priv-key"
+	ConfigPubKey      = "pub-key"
+	ConfigKeyId       = "key-id"
+	ConfigKeyPurposes = "key-purposes"
+)
+
+// getOpts iterates the inbound Options and returns a struct
+func getOpts(opt ...wrapping.Option) (*options, error) {
+	const op = "ed25519.getOpts"
+	// First, separate out options into local and global
+	opts := getDefaultOptions()
+	var wrappingOptions []wrapping.Option
+	var localOptions []OptionFunc
+	for _, o := range opt {
+		if o == nil {
+			continue
+		}
+		iface := o()
+		switch to := iface.(type) {
+		case wrapping.OptionFunc:
+			wrappingOptions = append(wrappingOptions, o)
+		case OptionFunc:
+			localOptions = append(localOptions, to)
+		}
+	}
+
+	// Parse the global options
+	var err error
+	opts.Options, err = wrapping.GetOpts(wrappingOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Don't ever return blank options
+	if opts.Options == nil {
+		opts.Options = new(wrapping.Options)
+	}
+
+	// Local options can be provided either via the WithConfigMap field
+	// (for over the plugin barrier or embedding) or via local option functions
+	// (for embedding). First pull from the option.
+	if opts.WithConfigMap != nil {
+		for k, v := range opts.WithConfigMap {
+			switch k {
+			case ConfigKeyId:
+				opts.WithKeyId = v
+			case ConfigKeyPurposes:
+				for _, raw := range strings.Split(v, ",") {
+					p, ok := wrapping.KeyPurpose_value[raw]
```

Comment:
What do you think about trimming spaces around the values too so users can use `purpose1, purpose2` safely?
```suggestion
					p, ok := wrapping.KeyPurpose_value[strings.TrimSpace(raw)]
```

Reply:
ty!